pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -343,16 +345,13 @@ func validateInstancesDistribution(ng *NodeGroup) error {
 }
 
 func validateNodeGroupSSH(SSH *NodeGroupSSH) error {
-	if SSH == nil {
```

Comment:
I think this check was good to have. Safer against null pointers.

Reply:
+1, however, is it expected to return `nil`?
Note that:

- with this check, then passing a `nil` `*NodeGroupSSH` **succeeds**,
- without this check, then passing a `nil` `*NodeGroupSSH` **fails** with a `nil` pointer error -- is it intended to no longer succeed?

Therefore, if we keep this check, should we keep `return nil` or should we return a proper error, e.g. `return errors.New("invalid NodeGroupSSH: nil")`?

Reply:
hmmm in the schema the NodeGroupSSH is optional so it's ok that it is nil

Reply:
@cPu1, what was the reason for removing this check? 🤔

Reply:
This check is now performed here: https://github.com/weaveworks/eksctl/pull/1530/files#diff-4572c83eda74aa47ee807019c6900162R159 before invoking this function.

Instead of having that function check if its argument is nil and then act on it, the caller performs the check and skips calling the function.

Reply:
Fair enough, either sounds good, given we only call `validateNodeGroupSSH` once anyway.